client/extensions/woocommerce/app/promotions/fields/promotion-applies-to-field/index.js

```diff
@@ -102,24 +104,66 @@ class PromotionAppliesToField extends React.Component {
 		this.initializeValue( appliesToType );
 	};
 
+	onProductIdChange = ( productId ) => {
+		const { edit } = this.props;
+		edit( 'appliesTo', { productIds: [ productId ] } );
+	};
+
+	onProductIdsChange = ( productIds ) => {
```

Comment:
`onProductIdChange` and `onProductIdsChange` could be a single function made with `lodash.concat`
`edit( 'appliesTo, { productids: concat( [], productIds } )`  productIds can be singular. Mentioning because I think this could further down simplify some logic like  `onChange={ ( singular ? this.onProductIdChange : this.onProductIdsChange ) }`